src/adaptix/_internal/morphing/concrete_provider.py

```diff
@@ -75,6 +75,57 @@ def datetime_format_dumper(data: datetime):
         return datetime_format_dumper
 
 
+@dataclass
+@for_predicate(datetime)
+class DatetimeTimestampProvider(LoaderProvider, DumperProvider):
+    tz: typing.Optional[timezone] = timezone.utc
+
+    def _provide_loader(self, mediator: Mediator, request: LoaderRequest) -> Loader:
+        tz = self.tz
+
+        def datetime_timestamp_loader(data):
+            try:
+                if data is None:
```

Comment:
Let's add some clarification here about why this guard is exist
```suggestion
                # Pure-Python implenentation and C-extension implementation
                # of datetime.date.fromtimestamp module works differently with a None arg.
                # See https://github.com/python/cpython/issues/120268 for more details.
                if data is None:
```

Reply:
> I think this is a good idea but `implenentation` contains typo

fixed